twarep/electrifly
# Soc flight graph

## Checklist
**Note:** Put an x into the box like this [x] to check these off

## What I Did

- Edited the `app.py` file to introduce the SOC vs. Time graph to the Data Analysis screen,
- Added multi-select functionality to the SOC vs. Time graph,
- Added dummy data to show the graph on the UI. The data is under the `test_data/` folder. Once the database is completely implemented with the correct Scaping for Firefox, I will change the graph to take data from it.
- Added the data tables from a previous PR. I had them on another file called `app_tables.py`, but it made more sense to add to `app.py`. They also use dummy data, which will be fixed later.

## How I Did it?
- For the graphing, I mostly used `matplotlib`, `numpy`, and `pandas`. Make sure you have those (The version doesn't matter fr this purpose).
- In terms of technicality, all the code is pretty self-explanatory.

## How to Test?
- Testing is simple. Just pull the changes. Then run the `app.py` script. Once Shiny is running, open the web page in a new browser and see if the first page you see is there:
![image](https://github.com/twarep/electrifly/assets/55631443/69afdad7-11d0-4bf0-a308-52f5cc8c0dae)

- From there, click the `Data Analysis` text. It should take you to here:
![image](https://github.com/twarep/electrifly/assets/55631443/25cbc092-237a-4b15-accd-24314c188b09)

- If you follow the instructions on the graph, you can select all the dates to show the data. It should look like this:
![image](https://github.com/twarep/electrifly/assets/55631443/9ff9c04b-5bd2-474c-922f-9317d5c22ade)


### Relevant Links
Notion Links:
1. https://www.notion.so/Analytics-Screen-Make-UI-SOC-Flight-Graph-f296d54deb33432c99c63cb8ae8564bd?pvs=4
2. https://www.notion.so/Thresholds-to-SOC-Graph-f83cd80e71fc4d3794114bdae82c7046?pvs=4
